src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx

```diff
@@ -68,28 +69,51 @@ export const K8SResourceList = ({
 
     const [resourceListLoader, _resourceList, _resourceListDataError, reloadResourceListData] = useAsync(
         () =>
-            abortPreviousRequests(
-                () =>
-                    getK8sResourceList(
+            abortPreviousRequests(async () => {
+                try {
+                    if (selectedResource.gvk.Kind === SIDEBAR_KEYS.nodeGVK.Kind) {
+                        const response = await getNodeList(clusterId, abortControllerRef.current.signal)
+
+                        return parseNodeList(response)
```

Comment:
Suggestion: Can we add version list api with this in Promise.allSettled if not handling error?, and nest the response { nodeK8sVersions, _resourceList }

Reply:
Not being able to fetch k8s versions should not prevent the user from viewing the page. But we can show this in the select picker. Will propagate this error to select picker.

Reply:
Even moved the logic to fetch the k8s versions into node search list